python/ray/data/dataset.py

```diff
@@ -522,6 +524,47 @@ def build_node_id_by_actor(actors: List[Any]) -> Dict[Any, str]:
             for actor in locality_hints
         ])
 
+    def split_at_indices(self, indices: List[int]) -> List["Dataset[T]"]:
```

Comment:
What do you think about following the same API as `np.split` and have this be `indices_or_sections` which can either be `int` or `List[int]`? I think splitting into N equal size chunks would be nice if it can be handled inside here directly.

Reply:
Hmm I'd prefer to not support it if unnecessary, since you can compute such a split by yourself dividing up ds.count().

Reply:
Yeah i'd lean towards keeping it in `split` to follow the numpy api, but see why we don't for now because of locality.

Btw I think list of indices is nice since it (1) follows the numpy convention (2) there are no surprises in edge case handling. I don't think it's obvious if padding vs truncation vs off by one are better behaviors and this is unambiguous